lib/pages/caregiver/rating_page.dart

```diff
@@ -54,12 +67,14 @@ class _CaregiverRatingPageState extends State<CaregiverRatingPage> {
 						  )
 				    ]
 					);
-        }
+				}
 			)
 		);
 	}
 
 	Widget _buildCarousel() {
+		WidgetsBinding.instance
+			.addPostFrameCallback((_) => _animateAfterReturn());
```

Comment:
- I'm not sure how costly `addPostFrameCallback` is but I would rather not schedule it every frame if not necessary. Can you put it directly in _returnCallback?
- `_animateAfterReturn` can be inlined into `addPostFrameCallback`
- Are you sure `addPostFrameCallback` is even required here? From my experience `onPressed` should be called outside of the widget build frame anyway..